include/zephyr/kernel_structs.h

```diff
@@ -32,6 +32,11 @@
 #include <zephyr/sys/rb.h>
 #endif
 
+#define NBITS (8 * sizeof(uint32_t))
```

Comment:
Move NBITS to the C file that uses it, or if there's more than one to a kernel-internal header somewhere.  This is a global header and tiny symbols with ambiguous names pollute the global namespace.

Also repeating the request that you make the type of the bitmask[] array be 64 bits on 64 bit platforms for efficiency.